pkg/activator/handler/handler.go

```diff
@@ -20,33 +20,48 @@ import (
 	"net/url"
 	"strconv"
 	"time"
-
 	"github.com/knative/serving/pkg/activator"
 	"github.com/knative/serving/pkg/activator/util"
 	pkghttp "github.com/knative/serving/pkg/http"
 	"go.uber.org/zap"
 )
 
+const overloadMessage = "throttler overload"
+
 // ActivationHandler will wait for an active endpoint for a revision
 // to be available before proxing the request
 type ActivationHandler struct {
 	Activator activator.Activator
 	Logger    *zap.SugaredLogger
 	Transport http.RoundTripper
 	Reporter  activator.StatsReporter
+	Throttler *activator.Throttler
 }
 
 func (a *ActivationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	namespace := pkghttp.LastHeaderValue(r.Header, activator.RevisionHeaderNamespace)
 	name := pkghttp.LastHeaderValue(r.Header, activator.RevisionHeaderName)
 
 	start := time.Now()
+	revID := activator.RevisionID{namespace, name}
+	a.Logger.Debug(revID)
+	a.Throttler.AddIfAbsent(revID)
+	// ActiveEndpoint() will block until the first endpoint is available
+	ar := a.Activator.ActiveEndpoint(namespace, name)
+	err := a.Throttler.Try(revID, func() {
+		a.activate(w, r, namespace, name, start, ar)
+	})
+	if err != nil {
```

Comment:
Is this the only reason why Try() would return an error?

Reply:
Reading below, I can see that it's not the case: it seems one is 500 the other case is 503?

Reply:
Breaker absence feels like a bug, so 500 is more suiting, IMO.

Reply:
I've decided to inline the breaker creation into Try and UpdateCapacity methods, so the error is not being returned by the method and we don't need to return a 500 to the user